### PR TITLE
Add button in contribution success page

### PR DESCRIPTION
**Pull request by nishant22029**
- Added short code to add Join as a volunteer button on contribution success page.

## Summary by CodeRabbit

- **New Features**
	- Introduced several new shortcodes for custom messages, volunteer registration buttons, and collection landing pages.
	- Enhanced user identification and volunteer registration functionalities with improved redirection logic.
	- Implemented a new validation script for better user experience and error handling during login.
	- Added a new query variable for improved URL parameter management.
	- Added responsive design classes for better layout flexibility across devices.
	- Introduced new SCSS and CSS classes for dynamic styling based on screen size.
	- Added new utility classes for widths, heights, and font sizes to enhance layout options.

- **Bug Fixes**
	- Improved error handling and validation during login and password reset processes.
	- Prevented induction emails for transitioned volunteers with completed "Material Contribution" activities.

### CODE REVIEW

File: wp-content/themes/goonj-crm/functions.php [hunk 1]
```diff
@@ -419,8 +419,63 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+	}
+	// Create the base URL for the volunteer form
+	$baseUrl = home_url('/volunteer-registration/form-with-details/');
+
+	$buttonUrl = esc_url(
+		$baseUrl . '#?' . http_build_query(array(
+			'Individual1' => $individualId,
+			'message' => 'individual-user'
+		))
+	);
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:462]:
```suggestion
	$redirectPathWithParams = $redirectPath . '#?' . http_build_query(array(
			'Individual1' => $individualId,
			'message' => 'individual-user'
		))
	);
```

File: wp-content/themes/goonj-crm/functions.php [hunk 2]
```diff
@@ -419,8 +419,63 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+	// Return the button HTML
+	return '
+		<div style="display: flex; justify-content: center; align-items: center; border-style:none; border-width:0px; border-radius:5px;">
+			<a class="wp-block-button__link has-white-color has-vivid-red-background-color has-text-color has-background has-link-color wp-element-button" 
+			   href="' . $buttonUrl . '" 
+			   style="border-style:none;border-width:0px;border-radius:5px">
+			   Wish to Volunteer ?
+			</a>
+		</div>';
+}
```
[the remark this answers is not in the document]
Comment (pokhiii) [reply wp-content/themes/goonj-crm/functions.php:473]: @nishant22029 Please implement this

**Review by pokhiii — CHANGES_REQUESTED** [review]

File: wp-content/civi-extensions/goonjcustom/Civi/InductionService.php [hunk 3]
```diff
@@ -216,6 +216,16 @@ public static function sendInductionEmailForTransitionedVolunteer(string $op, st
       return FALSE;
     }
 
+    // Prevent volunteer email activity for new users who contribute.
+    $isMaterialContributionActivityDone = Activity::get(FALSE)
+      ->addWhere('source_contact_id', '=', self::$transitionedVolunteerId)
+      ->addWhere('activity_type_id:label', '=', 'Material Contribution')
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/civi-extensions/goonjcustom/Civi/InductionService.php]: sure

File: wp-content/themes/goonj-crm/functions.php [hunk 4]
```diff
@@ -419,8 +419,63 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+	->execute();
+
+	$contactSubTypes = $contacts[0]['contact_sub_type'] ?? null;
+
+	// Check if contact_sub_type is not null or empty and contains 'Volunteer'
+	if (!empty($contactSubTypes) && in_array('Volunteer', $contactSubTypes)) {
+		return;
+	}
+	// Create the base URL for the volunteer form
+	$baseUrl = home_url('/volunteer-registration/form-with-details/');
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: Updated

File: wp-content/themes/goonj-crm/functions.php [hunk 5]
```diff
@@ -419,8 +419,63 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+		<div style="display: flex; justify-content: center; align-items: center; border-style:none; border-width:0px; border-radius:5px;">
+			<a class="wp-block-button__link has-white-color has-vivid-red-background-color has-text-color has-background has-link-color wp-element-button" 
+			   href="' . $buttonUrl . '" 
+			   style="border-style:none;border-width:0px;border-radius:5px">
+			   Wish to Volunteer ?
+			</a>
+		</div>';
+}
+
+add_shortcode('goonj_volunteer_registration_button', 'render_volunteer_button');
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: sure updated

File: wp-content/themes/goonj-crm/functions.php (hunk 1 above)
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: updated

File: wp-content/themes/goonj-crm/functions.php [hunk 6]
```diff
@@ -419,8 +419,67 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+
+    $activities = \Civi\Api4\Activity::get(FALSE)
+        ->addSelect('source_contact_id')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT')
+        ->addWhere('id', '=', $activityId)
+        ->addWhere('activity_type_id:label', '=', 'Material Contribution')
+        ->execute();
+
+    if ($activities->count() === 0) {
+        error_log('No activities found for Activity ID: ' . $activityId); // Log if no activities are found
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:449]: Use Civi::log

File: wp-content/themes/goonj-crm/functions.php [hunk 7]
```diff
@@ -419,8 +419,67 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+        return;
+    }
+
+    // Since we have activities, get the single record
+    $activity = $activities->single();
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:454]: If there are more than one activity then `single` will give error
```suggestion
    $activity = $activities->first();
```

File: wp-content/themes/goonj-crm/functions.php [hunk 8]
```diff
@@ -419,8 +419,67 @@ function goonj_is_volunteer_inducted( $volunteer ) {
 function goonj_custom_message_placeholder() {
 	return '<div id="custom-message" class="ml-24"></div>';
 }
+
 add_shortcode( 'goonj_volunteer_message', 'goonj_custom_message_placeholder' );
 
+function generate_volunteer_button_html($buttonUrl) {
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:425]: I've give this feedback earlier as well – any function in the theme's function file needs to be prefixed by `goonj_`

File: wp-content/themes/goonj-crm/functions.php [hunk 9]
```diff
@@ -419,8 +419,67 @@ function goonj_is_volunteer_inducted( $volunteer ) {
...
+    ));	
+
+    return generate_volunteer_button_html($redirectPathWithParams);
+}
+
+add_shortcode('goonj_contribution_volunteer_signup_button', 'goonj_contribution_volunteer_signup_button');
+
+
+
+
```
Comment (pokhiii) [inline wp-content/themes/goonj-crm/functions.php:482]: why so many empty lines?

**Review by pokhiii — CHANGES_REQUESTED** [review]

File: wp-content/themes/goonj-crm/functions.php (hunk 6 above)
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php:449]: sure

File: wp-content/themes/goonj-crm/functions.php (hunk 8 above)
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php:425]: updated with prefix

File: wp-content/themes/goonj-crm/functions.php (hunk 9 above)
Comment (nishant22029) [reply wp-content/themes/goonj-crm/functions.php]: removed empty lines

**Review by pokhiii — APPROVED** [review]

**Pull request merged**